projects/microphysics/scripts/score_training.py

```diff
@@ -40,6 +40,45 @@ def load_final_model_or_checkpoint(train_out_url) -> Tuple[tf.keras.Model, str]:
     return tf.keras.models.load_model(url_to_load), url_to_load
 
 
+def logit_to_one_hot(x):
+    return x == tf.math.reduce_max(x, axis=-1, keepdims=True)
+
+
+def score_gscond_classes(config, model, batch):
+    v = "gscond_classes"
+    if v not in model.output_names:
+        return {}, {}
+
+    transform = config.build_transform(batch)
+    batch_transformed = transform.forward(batch)
+    out = model(batch)
+
+    y = out[v]
+    predicted_class = logit_to_one_hot(y).numpy()
+    truth = batch_transformed[v].numpy()
+
+    profiles = {}
+    scalars = {}
+
+    for score in [vcm.accuracy, vcm.f1_score, vcm.precision, vcm.recall]:
+        profile = score(truth, predicted_class, mean=lambda x: x.mean(0))
```

Comment:
It's somewhat apparent by the variable names, but the lambda definitions are particular to the gscond_classes shape and may be hard to decode a few months down the line.  Either define the functions so the name is reflective of the average taken, or just add a comment somwehere depicting what the expected `gscond_classes` shape dimensions are.

Reply:
i added some comments.